src/histograms/AbstractHistogram.ts

```diff
@@ -263,6 +255,92 @@ export abstract class AbstractHistogram {
     }
   }
 
+  public getHorizontalOffset(chartAxes){
+    let h = this.chartDimensions.height;
+    if (isChartAxes(chartAxes)) {
+      if (!this.histogramParams.yAxisFromZero) {
+        const minMax = chartAxes.yDomain.domain();
+        if (minMax[0] >= 0) {
+          h = chartAxes.yDomain(minMax[0]);
+        } else {
+          h = chartAxes.yDomain(minMax[1]);
+        }
+      } else {
+        h = chartAxes.yDomain(0);
+      }
+    }
+    return h;
+  }
+
+  public updateNumberOfLabelDisplayedIfOverlap(chartAxes: ChartAxes | SwimlaneAxes, leftOffset = 0){
+    // Get the offset used when we will draw the labels.
+    const horizontalOffset = this.getHorizontalOffset(chartAxes);
+      let sumWidth = 0;
+      // create virtual nodes. Helps to get label's width.
+      const virtualLabels = this.chartDimensions.svg.append('g');
+      const labels = virtualLabels.append('g')
+          .attr('class', 'histogram__labels-axis')
+          .attr('transform', 'translate(' + (leftOffset - 1) + ',' + this.chartDimensions.height * this.histogramParams.xAxisPosition + ')')
+          .call(chartAxes.xLabelsAxis).selectAll('text');
+
+      // check for all label if there is an overlap.
+      let hasOverlap = false;
+      const nodes = labels.nodes();
+      for (let i = 0; i < labels.size(); i++) {
+        const next = i + 1;
+        if(nodes[next]){
+          const c = this.getDimension(nodes[i]);
```

Comment:
Do 'c' and 'n' stand for current and next ? if so, can you add a comment for those var names or rename the variable ?